### ISSUE TITLE
Mnemonic seed restores to wrong address with early blockchain

### ISSUE THREAD

**Issue by HuangYuSan** [monero-project/monero#3513]
I restored a Monero wallet from its mnemonic seed with `monero-wallet-cli`, while my local daemon was running in the background. However, I hadn't downloaded the blockchain yet, so the daemon was downloading the first few blocks at that point. A wrong address was restored from the seed, which was of course empty, but checking `seed` returned what I had entered.

I then copied the whole blockchain from another drive, restarted the daemon, and tried restoring the wallet again. At this point, it restored correctly. I then checked the other wallet, which now gave me a seed that was different from what I had entered, but contained some of the right words.

As recommended [here](https://monero.stackexchange.com/questions/7887/mnemonic-seed-restored-to-wrong-address-with-early-blockchain?noredirect=1#comment6729_7887), I can provide the two public addresses if needed.

**Comment by moneromooo-monero** [comment]
Are you sure you entered the same words in both cases ? The blockchain isn't used to parse words into a secret key thankfully.

**Comment by HuangYuSan** [comment]
Yes. When I checked the output of `seed` after parsing it was still correct, until I updated the blockchain. The distorted seed contains less than ten words from the real one, so it was quite obvious that it had changed. 

My theory was that maybe there was a change in behaviour of `monero-wallet-cli` at some point, and that the block height is used to determine whether the switch has happened yet.

**Comment by moneromooo-monero** [comment]
If you move the monero to a new address, are you able to then send me the seed for testing (you can GPG encrypt the data with my public key in monero's tree, in utils/gpg_keys/moneromooo.asc) ?

**Comment by HuangYuSan** [comment]
The wallet in question isn't mine, so I'd rather not, sorry. Anyway, here's the addresses if that helps!

-----BEGIN PGP MESSAGE-----

hQIMA3bOQo7+njNXARAAvOw+Uc8iEHH4pMycv3h+eKTjYKKytc/iGKM0farx/FI6
lKmnIuIlOXJjljM4tpLX5oFVAT46+Z8XN4ahBUNbQQo2HpGFFRaXfdbEUflR2yDg
q2ZM87R4qM3NEi6meTbOFi4ahUg5NWBLpF/hRkX55YHz1SSrqo9B1wy5lRvfJ1ET
izfLd5Dk5tFpfyEE0CnawOBcMzlRJb8ZRfnqtQLJgt/w4L5eH65g9iNrp0aIovDc
nLcoYOO6WOVqU9BM7jF4Q0jgw6KcCLtiREzMMcWihcuP6NUospxOWd/rE3Fjy4So
so7WXqtSFtFbBd3yc/dUkNqQ6zHm7vZwmZhSdTCxK8muxEgs44bKWyLidc4Dj8P0
x3v0bq+Ga5gXOXVA8V6WhfamOG+ZDVEJuMHuKIZ/EaomsEjgoxdopk1aStnDNoZF
0Xxk5f5QBM5a2qbFiZNxSurDld4UcLiFTWto1aMbBVEB5SuaoCaZRuwM29qynk09
SCg8EpheKy91j5cHarevScQojGDNml4pNluacDGj6FQsAsg8WzPrDhyUEbqqdzOS
mt6IW5416nQg5k6+MWqzt33J28dkNxT3EAmqaXpe76nLmEBsNOGfFqRWlkRbYyNN
uLei/DP1ivTJwwmGKH0xvGfpLQvCvn29doEHrlWxrm/mXNJeof/XRfg58dgE283S
wFIBlPofQBd+F1/lyQo2UQ7hTQZgjWVqi3vrgZ/gMb7yGKunlPM0G/wPCgiF1jRo
UxWwRRTQiv+1Tv5E/gQGdl0RDQEOUtQb5amSdpUzcvZTObkIOJ6tsOcomVM71M1C
sEehx7OWEgInnXebcvWIDHf2V43P5r5wbX4Q5mJtgaNdR2d62Ux+COwvR5bJue8g
RXIFlch1HyX6/IuCedIzlUHql+gYiljUQ+Zrflh4wnqwtNOZPG01kQnA/JIKkGCw
bXBFDThFKVB6pa003kFSt9LCL904Uw/d1JZHfprovWrAaKlzkzKHtjE/AHaSi7eo
0wKQ1/IZdf7mA+6bbot6PVn3X0KcxFrJ5/49An6iSR9Xq6to
=ts+9
-----END PGP MESSAGE-----

**Comment by moneromooo-monero** [comment]
OK. If you are able to, it would be nice if you could run this with valgrind. To do this, you (1) install valgrind, and (2) run your program with "valgrind " in front, eg: valgrind monero-wallet-cli --flag etc.
I'm interested in any output starting with ==. Alternatively, running with ASAN would also be OK, but that requires rebuilding after running cmake with -D SANITIZE=ON.
If you're able to build, and don't mind helping debug, and can still reproduce the issue, I might have some logging changes later on.

**Comment by HuangYuSan** [comment]
Ok, did that. I didn't get any memory errors, so no lines starting with '==' other than the init message. I can build if needed; I'll try reproducing the issue first.

**Comment by moneromooo-monero** [comment]
Could you reproduce the issue ?

**Comment by HuangYuSan** [comment]
Ah sorry! No, it worked fine afterwards. I definitely did enter the right seed though, I tried many times very carefully over the course of several days (since this was about a significant amount of money). Only after updating the blockchain it worked, and I checked with my paper slip afterwards, it definitely was correct on paper, and of the one that I had entered previously nearly all of the words had changed.
